src/common/telemetry/lib/Telemetry.cpp

```diff
@@ -0,0 +1,369 @@
+        {
+            OsConfigLogError(m_logfileHandle, "Exception during shutdown: %s", e.what());
+        }
+    }
+}
+
+bool TelemetryManager::ProcessJsonFile(const std::string& filePath)
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+
+    if (!m_initialized)
+    {
+        OsConfigLogError(m_logfileHandle, "TelemetryManager not initialized");
+        return false;
+    }
+
+    FILE* file = fopen(filePath.c_str(), "r");
+    if (!file)
+    {
+        OsConfigLogError(m_logfileHandle, "Failed to open file: %s", filePath.c_str());
+        return false;
+    }
+
+    char* line = nullptr;
+    size_t lineSize = 0;
+    ssize_t lineLength;
+    bool success = true;
+
+    try
+    {
+        // Read file line by line
+        while ((lineLength = getline(&line, &lineSize, file)) != -1)
+        {
+            // Remove trailing newline if present
+            if (lineLength > 0 && line[lineLength - 1] == '\n')
+            {
+                line[lineLength - 1] = '\0';
+            }
+
+            // Skip empty lines
+            if (strlen(line) == 0)
+            {
+                continue;
+            }
+
+            // Process the JSON line
+            ProcessJsonLine(std::string(line));
+        }
+    }
+    catch (const std::exception& e)
+    {
+        OsConfigLogError(m_logfileHandle, "Error processing JSON file '%s': %s", filePath.c_str(), e.what());
+        success = false;
+    }
+
+    // Cleanup
+    if (line)
+    {
+        free(line);
+    }
+    fclose(file);
+
+    return success;
+}
+
+bool TelemetryManager::ValidateEventParameters(const std::string& eventName, const std::set<std::string>& jsonKeys) const
+{
+    auto it = EVENT_PARAMETER_SETS.find(eventName);
+    if (it == EVENT_PARAMETER_SETS.end())
+    {
+        OsConfigLogError(m_logfileHandle, "Unknown event type: %s", eventName.c_str());
+        return false;
+    }
+
+    const auto& requiredParams = it->second.first;
+    const auto& optionalParams = it->second.second;
+
+    // Check that all required parameters are present
+    for (const auto& requiredParam : requiredParams)
+    {
+        if (jsonKeys.find(requiredParam) == jsonKeys.end())
+        {
+            OsConfigLogError(m_logfileHandle, "Missing required parameter '%s' for event '%s'", requiredParam.c_str(), eventName.c_str());
+            return false;
+        }
+    }
+
+    // Check that no unexpected parameters are present
+    for (const auto& jsonKey : jsonKeys)
+    {
+        if (jsonKey == "EventName") continue; // Skip the event name field
+
+        if (requiredParams.find(jsonKey) == requiredParams.end() &&
+            optionalParams.find(jsonKey) == optionalParams.end())
+        {
+            OsConfigLogError(m_logfileHandle, "Unexpected parameter '%s' for event '%s'", jsonKey.c_str(), eventName.c_str());
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void TelemetryManager::ProcessJsonLine(const std::string& jsonLine)
```

Comment:
>TelemetryManager::ProcessJsonLine [](http://example.com/codeflow?start=5&length=33)

Why is this not validating that the TelemetryManager is initialized?

Please make those members private and invoke them from inside this function, at beginning and at the end,